GoogleCloudPlatform/cluster-toolkit
# Add max_pods_per_node for GKE cluster and nodepool

### Submission Checklist
Aim of this PR is to allow specifying max_pods_per node in GKE cluster and node pool. The change is introduced so user can specify number of pods per node and meet network requirements in specific cases.

Manual tests:
- specify default_max_pods_per_node and max_pods_per_node in cluster and check if it is applied
- specify default_max_pods_per_node and networking_mode to ROUTES and check if precondition fails

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)
